pkg/apis/kf/v1alpha1/app_types.go

```diff
@@ -117,6 +124,22 @@ type AppSpecInstances struct {
 	Max *int `json:"max,omitempty"`
 }
 
+// AppSpecServiceBinding is a binding to an external service.
+type AppSpecServiceBinding struct {
+
+	// InstanceRef is the service the app will bind to.
+	InstanceRef servicecatalogv1beta1.LocalObjectReference `json:"instanceRef"`
+
+	// Parameters is an arbitrary JSON to be injected into VCAP_SERVICES.
+	// +optional
+	Parameters []byte `json:"parameters,omitempty"`
+
+	// BindingName is the name of the binding.
+	// If unspecified it will default to the service name
+	// +optional
+	BindingName string `json:"bindingName,omitempty"`
```

Comment:
I think this should be set by the defaulting webhook to be the value from `InstanceRef` and make a validator that requires BindingName to be set.

Then we could use it as a primary key to ensure users don't try to create two bindings with the same name.